rust/stackable-cockpit/src/engine/kind/mod.rs

```diff
@@ -1,46 +1,57 @@
 use std::process::Stdio;
 
-use snafu::{ensure, ResultExt, Snafu};
+use snafu::{ensure, OptionExt, ResultExt, Snafu};
 use tokio::{io::AsyncWriteExt, process::Command};
 use tracing::{debug, info, instrument};
 
 use crate::{
     constants::DEFAULT_LOCAL_CLUSTER_NAME,
-    engine::{check_if_docker_is_running, kind::config::KindClusterConfig, DockerError},
+    engine::{
+        docker::{self, check_if_docker_is_running},
+        kind::config::Config,
+    },
     utils::check::binaries_present_with_name,
 };
 
 mod config;
 
+type Result<T, E = Error> = std::result::Result<T, E>;
+
 #[derive(Debug, Snafu)]
-pub enum KindClusterError {
-    #[snafu(display("io error"))]
-    IoError { source: std::io::Error },
+pub enum Error {
+    #[snafu(display("failed to pipe kind config using stdin"))]
+    PipeConfigStdin { source: std::io::Error },
+
+    #[snafu(display("failed to obtain stdin handle"))]
+    ObtainStdinHandle,
 
-    #[snafu(display("no stdin error"))]
-    StdinError,
+    #[snafu(display("failed to start kind command"))]
```

Comment:
Kind _eventually_ needs to be in title case, but I wouldn't block the PR for that.

Reply:
Yes. Will open a follow-up PR for that. Let's move for now.